src/debugging/Inspector/scripts/generate.py

```diff
@@ -0,0 +1,197 @@
+    elif isinstance(parameter_type, PrimitiveType): 
+        raw_name = parameter_type.raw_name()   
+        ts_name = _PRIMITIVE_TO_TS_NAME_MAP.get(raw_name)
+        type_raw_name = ts_name if ts_name else raw_name
+
+    return type_raw_name
+
+class TypeScriptInterfaceGenerator(Generator):
+    def __init__(self, model):
+        Generator.__init__(self, model, "")
+    
+    def output_filename(self):
+        return "InspectorBackendCommands.ts"
+
+    def generate_output(self):
+        sections = []
+        sections.append('''interface FrontendDispatcher { 
+    dispatch(message: string): void; 
+}
+        ''')
+        sections.append('''class Dispatcher {
+    dispatch(method: string, argumentsMessage: string) {
+        var json = JSON.parse(argumentsMessage);
+        var methodArguments = [];        
+        var params = json.params;
+        for (var property in params) {
+            var methodArgument = params[property];
+
+            methodArguments.push(methodArgument);
+        }
+
+        if (this[method] != null) {
+            var result = this[method].apply(this, methodArguments);
+            return JSON.stringify(result);
+        } else {
+            throw new Error("Method " + method + " not found");
+        }
+    }
+}''')
+        sections.extend(map(self.generate_domain, Generator.domains_to_generate(self)))
+        return "\n\n".join(sections)
+
+    def generate_domain(self, domain):
+        lines = []
+        lines.append('// %s.' % domain.domain_name)
+        lines.append('namespace %sDomain {' % domain.domain_name)
+
+        lines.extend(self.generate_domain_type_declarations(domain))
+        
+        lines.append("export interface %sDomainDispatcher { " % domain.domain_name);
+        lines.extend(self.generate_domain_commands(domain))
+        lines.append('}')
+
+        lines.append("export class %sFrontend { " % domain.domain_name);
+        lines.append('\tconstructor(private dispatchMessage: (message: String) => void) { \n\t}')
+        lines.extend(self.generate_domain_events(domain))
+        lines.append('}')
+
+        lines.append('}')
+        return '\n'.join(lines)
+
+    def generate_domain_type_declarations(self, domain):
+        lines = []
+        primitive_declarations = filter(lambda decl: isinstance(decl.type, AliasedType), domain.type_declarations)
+        object_declarations = filter(lambda decl: isinstance(decl.type, ObjectType), domain.type_declarations)
+        enum_declarations = filter(lambda decl: isinstance(decl.type, EnumType), domain.type_declarations)
+
+        for declaration in primitive_declarations:
+            description = declaration.description if declaration.description else 'No description'  
+            lines.append('export type {} = {} // {}'.format(declaration.type_name, ts_name_for_primitive_type(domain, declaration.type.aliased_type), description))
```

Comment:
Just a suggestion - you could put the description comment before the declaration, so that the code completion finds it. Also in the generated JS sources, the description comes at the end of the method and looks a bit confusing.